### PR TITLE
Upgrade to C++20

### LINKED ISSUE #54: [FEA] C++20
**Is your feature request related to a problem? Please describe.**

Enable C++20

**Describe the solution you'd like**

https://github.com/nv-morpheus/SRF/issues/54#tasklist-block-6dd488d0-7bd5-47a5-a727-4785f8399997

### LINKED ISSUE #231: [BUG]: clang version not updating in CI
### Version

23.01

### Which installation method(s) does this occur on?

_No response_

### Describe the bug.

The version of clang is baked into the NGC container CI runner image.

The `update_conda_env()` function in `ci/scripts/github/common.sh` only updates the `dev_env.yml` environment.

### Minimum reproducible example

```shell
Commit: https://github.com/nv-morpheus/MRC/pull/230/commits/d6985ae3a9c9474206f58f3df76faa3943544977
```

### Relevant log output

_No response_

### Full env printout

_No response_

### Other/Misc.

_No response_

### Code of Conduct

- [X] I agree to follow MRC's Code of Conduct
- [X] I have searched the [open bugs](https://github.com/nv-morpheus/MRC/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by ryanolson**
Closes #54
Closes #231

### CODE REVIEW

**Comment by ryanolson** [comment]
rerun tests

**Comment by pdmack** [comment]
rerun tests

**Comment by ryanolson** [comment]
rerun tests

File: include/mrc/core/spin_lock_mutex.hpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ namespace mrc::core {
 class SpinLockMutex
 {
   public:
-    // todo(clang-format-14)
+    // todo(clang-format-15)
```
Comment (mdemoret-nv) [inline include/mrc/core/spin_lock_mutex.hpp:27]: Shouldnt this just be removed?

File: include/mrc/memory/buffer_view.hpp [hunk 2]
```diff
@@ -39,7 +39,7 @@ class const_buffer_view  // NOLINT
     const_buffer_view()          = default;
     virtual ~const_buffer_view() = default;
 
-    // todo(clang-format-14)
+    // todo(clang-format-15)
```
Comment (mdemoret-nv) [inline include/mrc/memory/buffer_view.hpp:42]: Shouldnt this just be removed?

File: src/internal/network/resources.hpp [hunk 3]
```diff
@@ -40,7 +40,7 @@ class Resources final : private resources::PartitionResourceBase
 
     DELETE_COPYABILITY(Resources);
 
-    // todo(clang-format-14)
+    // todo(clang-format-15)
```
Comment (mdemoret-nv) [inline src/internal/network/resources.hpp:43]: Shouldnt this just be removed?

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
Couple of changes needed.

File: include/mrc/memory/buffer_view.hpp (hunk 2 above)
Comment (ryanolson) [reply include/mrc/memory/buffer_view.hpp:42]: Will be done with #219

File: Dockerfile [hunk 4]
```diff
@@ -133,7 +133,7 @@ RUN --mount=type=cache,id=conda_pkgs,target=/opt/conda/pkgs,sharing=locked \
 RUN source activate ${CONDA_ENV_NAME} &&\
     git clone https://github.com/include-what-you-use/include-what-you-use.git /opt/iwyu &&\
     cd /opt/iwyu &&\
-    git checkout clang_12 &&\
+    git checkout clang_14 &&\
```
[the remark this answers is not in the document]
Comment (ryanolson) [reply Dockerfile]: This file is deleted.

**Review by mdemoret-nv — APPROVED** [review]

**Comment by mdemoret-nv** [comment]
@gpucibot merge

**Pull request merged**